internal/cmd/beta/cdn/distribution/update/update_test.go

```diff
@@ -0,0 +1,365 @@
+	for _, m := range mods {
+		m(p)
+	}
+	return p
+}
+
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description string
+		argValues   []string
+		flagValues  map[string]string
+		isValid     bool
+		expected    *inputModel
+	}{
+		{
+			description: "base",
+			argValues:   []string{testDistributionID},
+			flagValues:  fixtureFlagValues(),
+			isValid:     true,
+			expected:    fixtureInputModel(),
+		},
+		{
+			description: "distribution id missing",
+			argValues:   []string{},
+			flagValues:  fixtureFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "invalid distribution id",
+			argValues:   []string{"invalid-uuid"},
+			flagValues:  fixtureFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "project id missing",
+			argValues:   []string{testDistributionID},
+			flagValues:  fixtureFlagValues(func(m map[string]string) { delete(m, globalflags.ProjectIdFlag) }),
+			isValid:     false,
+		},
+		{
+			description: "invalid distribution id",
+			argValues:   []string{"invalid-uuid"},
+			flagValues:  fixtureFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "both backends",
+			argValues:   []string{testDistributionID},
+			flagValues: fixtureFlagValues(
+				func(m map[string]string) {
+					m[flagHTTP] = "true"
+					m[flagBucket] = "true"
+				},
+			),
+			isValid: false,
+		},
+		{
+			description: "max config without backend",
+			argValues:   []string{testDistributionID},
+			flagValues: fixtureFlagValues(
+				func(m map[string]string) {
+					m[flagRegions] = "EU,US"
+					m[flagBlockedCountries] = "DE,AT,CH"
+					m[flagBlockedIPs] = "127.0.0.1,10.0.0.8"
+					m[flagDefaultCacheDuration] = "P1DT12H"
+					m[flagLoki] = "true"
+					m[flagLokiUsername] = "loki-user"
+					m[flagLokiPushURL] = "https://loki.example.com"
+					m[flagMonthlyLimitBytes] = fmt.Sprintf("%d", testMonthlyLimitBytes)
+					m[flagOptimizer] = "true"
+				},
+			),
+			isValid: true,
+			expected: fixtureInputModel(
+				func(m *inputModel) {
+					m.Regions = []cdn.Region{cdn.REGION_EU, cdn.REGION_US}
+					m.BlockedCountries = []string{"DE", "AT", "CH"}
+					m.BlockedIPs = []string{"127.0.0.1", "10.0.0.8"}
+					m.DefaultCacheDuration = "P1DT12H"
+					m.Loki = &lokiInputModel{
+						Username: "loki-user",
+						PushURL:  "https://loki.example.com",
+					}
+					m.MonthlyLimitBytes = utils.Ptr(testMonthlyLimitBytes)
+					m.Optimizer = utils.Ptr(true)
+				},
+			),
+		},
+		{
+			description: "max config http backend",
+			argValues:   []string{testDistributionID},
+			flagValues: fixtureFlagValues(
+				func(m map[string]string) {
+					m[flagHTTP] = "true"
+					m[flagHTTPOriginURL] = "https://origin.example.com"
+					m[flagHTTPOriginRequestHeaders] = "X-Example-Header: example-value, X-Another-Header: another-value"
+					m[flagHTTPGeofencing] = "https://dach.example.com DE,AT,CH"
+				},
+			),
+			isValid: true,
+			expected: fixtureInputModel(
+				func(m *inputModel) {
```

Comment:
```suggestion
				func(model *inputModel) {
```
same here, I would prefer proper names if possible

Reply:
Applies to all tests in this table test